### ISSUE TITLE
"When Idle" does not work for Automatic export

### ISSUE THREAD

**Issue by cjpoor** [retorquere/zotero-better-bibtex#2105]
support log ID: SUS6NR6D-refs-apse
My question does not relate to items in my library at all, so I just selected a random item in my library.

My problem is that automatic export of .bib file keeps occurring while I am using Zotero even though I have "When Idle" chosen as the option of when to export.
I think this may be causing text that I type into "info" for items to be deleted.
I'm on Linux Mint.

Many thanks, 
Chris.

**Comment by retorquere** [comment]
I haven't forgotten about this, I'm just working on a change I need to finish before picking up other stuff

**Comment by cjpoor** [comment]
Thanks, Emiliano.
Keep up the great work.

**Comment by glocalglocal** [comment]
Coming from #2112, 6.4.3.2397 doesn't seem to make a difference here. Exporting remains hyperactive. What makes things worse is that BBT seems to queue exports, so as I carry on editing one or more entries, BBT wakes up and performs multiple cycles of Preparing/Exporting in quick succession causing worse freezes.

**Comment by retorquere** [comment]
It just adds logging for diagnosis, it's not expected to make a difference yet, I don't know what causes it yet. I'll need a new log from the new build.

**Comment by cjpoor** [comment]
OK. Installed test build. Once again, clicking on a random entry:  Here's the report ID: S8PYHKI9-refs-apse

**Comment by retorquere** [comment]
A new build will drop soon for more logging, but as far as I can tell, `S8PYHKI9-refs-apse` shows BBT holding jobs until idle (except one error job for which the target directory is missing). The new build will give me more detail though.

**Comment by cjpoor** [comment]
New support log ID: TALIKPKV-refs-apse

**Comment by cjpoor** [comment]
Yes. I think you might have fixed it.
Thanks.
 I'll keep checking.

**Comment by retorquere** [comment]
Strange. I only added logging.

**Comment by cjpoor** [comment]
Ah. I need to make more observations.

**Comment by glocalglocal** [comment]
I haven't installed this test build to avoid mixing it with another test build. Am I right thinking the focus here is on 'When idle' not making a difference? I am asking because in #2112 backlinked above I also reported this:

>To make things worse, it looks like an export is triggered even when the field I am editing would not affect the Citation Key and the change should ideally be ignored.

Eg why would editing the Abstract trigger an export? If this is unrelated, perhaps I should create a separate issue.

**Comment by retorquere** [comment]
Because the relationship between the item and the exported bibtex entry is the result of decisions made in hundreds of not thousands of lines of code, and predicting whether an item should be written anew is exactly as complex as writing it?

**Comment by glocalglocal** [comment]
> predicting whether an item should be written anew is exactly as complex as writing it

In that case, is there an efficient way to merge queued updates? Here is a scenario: I have ten items I marked as related. Zutilo allows bulk relating but AFAIK not bulk unrelating. So, I changed my mind and I had to click the (-) button next to each one of the 9 related item for each one of the 10 items. That was annoying enough, but it was made worse by BBT constantly freezing Zotero to do export after export. If 'When idle' worked, BBT at least wouldn't get in my way, but I suspect it would still execute all those queued cycles of exporting. If my suspicion is correct, hopefully, these cycles could be merged into one cycle to be executed 'when idle'.

**Comment by retorquere** [comment]
What you describe is already how bbt works; autoexportdelay (I'm not at my computer but I think it's 5 seconds by default) merges such updates when auto export is immediate, and with on idle it's not supposed to be active when you are at all. If it doesn't for you that's a bug I must fix.

If you have immediate try setting the delay higher. If it's on idle, zotero should not be idle when you're clicking stuff.

**Comment by glocalglocal** [comment]
>with on idle it's not supposed to be active when you are at all. If it doesn't for you that's a bug I must fix. 

I am on 6.5.1 and 'when idle' still doesn't work. Like before, I get multiple exports in quick succession as I am editing items -- while typing even, which results in missed keystrokes and skipped letters. On a slower machine I would switch exporting on/off manually. Actually, I may do this on this machine for batch edits.

>When zotero comes out of idle, each auto export should be executed exactly once, and if not, that's a bug.

Good to hear though that exporting is meant to happen once. I think after I stop clicking/typing BBT exports more than once, but I am not sure because exports blend into each other as long as 'when idle' doesn't work.

**Comment by retorquere** [comment]
@glocalglocal turn on debug logging in the help menu, reproduce the problem, and send a log, then I can take a look.

**Comment by glocalglocal** [comment]
Done: D2060941578

I just edited the Abstract for the same item 3-4 times.

**Comment by retorquere** [comment]
I need a BBT log. D logs are sent to Zotero, I have no access to those.

**Comment by glocalglocal** [comment]
Oops. EJ97SBLF-euc

**Comment by cjpoor** [comment]
I can confirm that BBT is still exporting when I am working in Zotero. Support log ID: WC4TIHT7-refs-apse
By the way @glocalglocal do you use the Zotero abstract cleaner plug-in?

**Comment by retorquere** [comment]
I don't see Zotero going idle or *any* exports (auto or otherwise) happening in `WC4TIHT7-refs-apse`.

**Comment by retorquere** [comment]
`WC4TIHT7-refs-apse` is from 6.5.1, you must have upgraded from the build that dropped here before. A new 6.5.1-based build will drop here in 10 minutes or so. I'll need a new log from that.

**Comment by glocalglocal** [comment]
@retorquere here is the log ID with test build 6.5.1.2457: X3RJTL96-euc 

@cjpoor no, I don't have the Zotero abstract cleaner plug-in. I never needed to clean the abstracts. A bulk unrelate plug-in would be great though.

**Comment by glocalglocal** [comment]
>Install in Zotero by downloading [test build 6.5.1.2458](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-6.5.1.2458.xpi), 

There you are: MQ72G7KP-euc

**Comment by retorquere** [comment]
I pause auto-export during sync, as sync can bring a spate of updates, but I think end-of-sync currently always causes a resume, regardless of idle state. A new build will drop in 10 minutes, if that was the problem, this build should fix it.

**Comment by glocalglocal** [comment]
With .2468 BBT is permanently stuck in this state: 
![image](https://user-images.githubusercontent.com/16408198/164793894-06058252-694c-452e-b44a-5e3a2f7e9de5.png)

It is like that from the moment Zotero starts and BBT doesn't seem to do an autoexport ever.

**Comment by retorquere** [comment]
Does log generation still work on 2468? I'm not seeing this stuck behavior, and it has passed the test suite, or it wouldn't drop here. Auto-export works for me, too.

**Comment by glocalglocal** [comment]
I reinstalled 2468 to generate a log: MNUY5Y4R-euc

One correction: BBT gets stuck in that state when the first export is triggered -- not the moment Zotero starts. 

> edit: if you're reverting to 2458 you might as well just revert to the main release. 2458 has no changes against the main release except more logging.

I know. I just had to go back to something that worked because I am actually using Zotero + BBT + Obsidian for my work.

**Comment by retorquere** [comment]
No that's fine, I actually did mean literally that if you need something that works, I would recommend going to a formal release.

**Comment by retorquere** [comment]
Bizarre. No errors in `MNUY5Y4R-euc`, and all activity I see is that auto-exports are being held off because your system isn't idle.

> One correction: BBT gets stuck in that state when the first export is triggered -- not the moment Zotero starts.

Like when an auto-export is triggered?

**Comment by retorquere** [comment]
> I know. I just had to go back to something that worked because I am actually using Zotero + BBT + Obsidian for my work.

BTW no obligation, but you don't actually have to disrupt your work profile for stuff like this: https://www.zotero.org/support/kb/multiple_profiles

**Comment by retorquere** [comment]
@cjpoor do you see the same problem @glocalglocal has? I can't reproduce the last problem

**Comment by retorquere** [comment]
@glocalglocal could it be you're auto-exporting a large library? You have caching turned off, that could be part of the problem.

**Comment by glocalglocal** [comment]
>Bizarre. No errors in MNUY5Y4R-euc, and all activity I see is that auto-exports are being held off because your system isn't idle.

So, previously it thought the system was idle when it wasn't, and now it thinks it isn't idle when it is. Hmm. I am assuming that 'idle' means not doing anything in Zotero. In any case, to be sure I sat back doing nothing for a while and BBT remained in that state. 

>Like when an auto-export is triggered?

yes. The first auto-export puts BBT in that state.

**Comment by glocalglocal** [comment]
> BTW no obligation, but you don't actually have to disrupt your work profile for stuff like this: https://www.zotero.org/support/kb/multiple_profiles. That's how I use it myself.

Good tip. I will look into it. Thanks.

**Comment by retorquere** [comment]
> So, previously it thought the system was idle when it wasn't, and now it thinks it isn't idle when it is. Hmm. I am assuming that 'idle' means not doing anything in Zotero.

It means you aren't doing anything with your computer: https://udn.realityripple.com/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIIdleService:

> The idle service is for computer-wide idle detection, not just application idle detection. In other words, even if the user is working in other applications, the idle service will still consider the user to be active.

> yes. The first auto-export puts BBT in that state.

I'll try this with my large-library test with caching off. I haven't been able to reproduce it yet.

**Comment by glocalglocal** [comment]
> @glocalglocal could it be you're auto-exporting a large library? You have caching turned off, that could be part of the problem.

Is 3500 items a lot? My database is around 500MB. I think it's relatively light because I only link to files, never actually attach them.  

I didn't even know about caching. I'll go and find it. I am not sure why it's off.

**Comment by retorquere** [comment]
> So, previously it thought the system was idle when it wasn't,

I'm not 100% sure what you experienced previously, but I think the most likely is that you have auto-sync on, and Zotero reports every separate iem being synced as sync-on, sync-off events. I was resuming the exports on sync-off. Still should not have caused multiple exports unless it had unfortunate timing where each change was just over the buffer time apart. BBT previously only listened to idle-on/idle-off events, but then when sync-off came, it didn't check for the idle state and just resumed. I'm testing for the idle state now.

> Is 3500 items a lot? My database is around 500MB. I think it's relatively light because I only link to files, never actually attach them.

No, not really, my large-library test is 24k items. Attachments don't matter for auto-export as auto-exports are always without attachment export.

> I didn't even know about caching. I'll go and find it. I am not sure why it's off.

Preferences - BBT - Export - Misc - checkbox near the bottom.

**Comment by retorquere** [comment]
> > edit: if you're reverting to 2458 you might as well just revert to the main release. 2458 has no changes against the main release except more logging.
> 
> I know. I just had to go back to something that worked because I am actually using Zotero + BBT + Obsidian for my work.

My comment wasn't meant sarcastic/snarky, but I can see why it could have been read that way. Apologies.

**Comment by retorquere** [comment]
still not reproducible with 24k without caching, so it's not simply a performance problem.

**Comment by glocalglocal** [comment]
> Preferences - BBT - Export - Misc - checkbox near the bottom.

Caching is on now. I am guessing this means BBT won't 'prepare' the whole database every time. I am fairly certain I never switched this off. 

No difference with 2468 though. Not surprising because it doesn't even seem to try to initiate an export. Earlier versions where exporting didn't stick will probably perform faster exports now.

**Comment by cjpoor** [comment]
> @cjpoor do you see the same problem @glocalglocal has? I can't reproduce the last problem

What I saw just now was a delay of about 90 seconds before BBT prepared exports.
Support log ID: JABFTQ8U-refs-apse

**Comment by retorquere** [comment]
> Caching is on now. I am guessing this means BBT won't 'prepare' the whole database every time. I am fairly certain I never switched this off. 

The prepare should always happen, unless background export is off.

**Comment by retorquere** [comment]
> What I saw just now was a delay of about 90 seconds before BBT prepared exports.

The delay should be 127 seconds in your case, so if you counted out seconds without help by a stopwatch, that 90 estimate could well be right.

**Comment by cjpoor** [comment]
Right. So “when idle” appears to be working?

**Comment by glocalglocal** [comment]
I said: 
>I didn't even know about caching. I'll go and find it. I am not sure why it's off.

I think I know now why it was off. I switch caching on, and every time I restart Zotero I see it's always off. Reproducible every time. Also, I always see `Cached: ?%`

I also noticed that the frozen BBT doughnut appears and then the Zotero sync icon is unusually active -- ie the Zotero sync icon seems to come alive in under a second after I edit a field.   

Is there a key combination to force 'Export Now'? I could switch auto-export off and press a key combination to 'Export Now'. Another possibility could be showing the BBT doughnut all the time when auto-export is on, and we could click it to 'Export Now' (like the Zotero sync icon works).

**Comment by retorquere** [comment]
> I said:
> 
> > I didn't even know about caching. I'll go and find it. I am not sure why it's off.
> 
> I think I know now why it was off. I switch caching on, and every time I restart Zotero I see it's always off. Reproducible every time. 

Well that's not great. I'll look into that.

> Also, I always see `Cached: ?%`

That updates after it has exported once with caching on.

> I also noticed that the frozen BBT doughnut appears and then the Zotero sync icon is unusually active -- ie the Zotero sync icon seems to come alive in under a second after I edit a field.

That's part of the problem. I react to sync because I do not want auto-exports to be active during sync. But if sync comes in as a flurry of small one-item syncs, that would cause a flurry of queue resumes. Under immediate that should be caught under auto-export delay, under idle that should be caught by idle-wait. But the latter wasn't the case in production builds.

> Is there a key combination to force 'Export Now'? I could switch auto-export off and press a key combination to 'Export Now'. Another possibility could be showing the BBT doughnut all the time when auto-export is on, and we could click it to 'Export Now' (like the Zotero sync icon works).

I don't do shortcuts, that's Zutilo's domain. A clickable icon... I'll give that a peek, I haven't animated buttons before. You can currently kick of collection auto-export by right-clicking the collection.

**Comment by retorquere** [comment]
> EDIT: It's official: My screensaver comes on, which means the system has been idle for 4 mins. The frozen BBT is still there.

I'll need a log unfortunately to be able to tell what's going on, since I can't reproduce the problem on my end.

**Comment by glocalglocal** [comment]
2UA7HF23-euc

Here is what I did: 
I switched on caching (again), did 'Export Now'. 'Cached: ?%' became 'Cached: 0%'.
Started logging and restarted Zotero. 

Zotero completed its startup sync. 
I edited an item. Zotero sync immediately kicked off. BBT then followed immediately and completed successfully. Then I edited the item again. Zotero sync immediately started/finished again, and BBT appeared in that frozen state. I waited 4 mins and nothing happened. 

I checked caching and it was off! 
I sent you the log.

**Comment by cjpoor** [comment]
| I switch caching on, and every time I restart Zotero I see it's always off. Reproducible every time.
I observe the same issue. Enable caching for background exports gets unticked upon Zotero restart.
Support log ID: YUQP3K2F-refs-apse

**Comment by retorquere** [comment]
Cache was on at the time `YUQP3K2F-refs-apse` was generated. `2UA7HF23-euc` has it off. I'm trying to get this reproducible.

**Comment by cjpoor** [comment]
This cache unticking does not happen on my Mac.

**Comment by cjpoor** [comment]
Done.
Better BibTex preference: Export>Miscellaneous>Enable caching for background exports 
now remains ticked on restart of Zotero on Linux Mint
Latest Support log ID: RNZG24WF-refs-apse

**Comment by retorquere** [comment]
OK, so that's one problem sorted. The auto-export then, @cjpoor, should there be one in `RNZG24WF-refs-apse`? If not, can you reproduce the on-idle problem and send a new log?

**Comment by glocalglocal** [comment]
Export caching stays on for me as well. I also see 'Cached: 50%' -- I 
was seeing 0% previously.

**Comment by retorquere** [comment]
Weird thing is now one of my cache tests fail.

**Comment by cjpoor** [comment]
BBT automatic export now appears to only occur when idle. Also does not seem to interfere with editing of Zotero item.
Latest Support log ID following auto-export: JD99A4VT-refs-apse

**Comment by cjpoor** [comment]
On Macbook (albeit running High Sierra):
BBT now seems to retain ticked "Enable caching for background exports" erratically (only sometimes).
Also (on Mac), with Automatic export set to "On Change" or "When idle" the export is no longer being done (frozen in initial state) following addition of item to Zotero (although this triggers sync).
Support log ID: UL3TXZ6M-refs-apse

**Comment by retorquere** [comment]
the cache tests pass again.

**Comment by glocalglocal** [comment]
With 2481 caching stays on when I restart Zotero but after Export Now, 
'Cached:' changed from ?% to 50% and later to ?%. It remains ?% even 
after I tried Export Now several times.

The BBT icon again stays on forever, like it's waiting for something 
before it starts auto-exporting.

**Comment by retorquere** [comment]
*sigh* here we go again.

**Comment by cjpoor** [comment]
Still no auto export on Mac with .2484
After Export now, Cached: ?%

**Comment by retorquere** [comment]
Can I get a log again? If the cache rate is `?` that means there's an error that will have been logged.

**Comment by retorquere** [comment]
The absolute bonkers thing about this is that I have an auto-export test in my suite. Builds wouldn't drop here if that test failed. I cannot drop builds that have a single failing test.

**Comment by cjpoor** [comment]
I think caching must be taking place because the export is much quicker than the one I did earlier (the first for some time).
Although, strangely, I just did a number of "Export now"s, some were very quick but the last one wasn't.
Support log ID before .2488: 7MCQ82YT-refs-apse

**Comment by cjpoor** [comment]
Support log ID with .2488: 762ZL8G2-refs-apse

**Comment by cjpoor** [comment]
Support log ID with .2489: VPZ2M7ZP-refs-apse
I will be away from my computer for a while.

**Comment by retorquere** [comment]
I see no attempt to refresh the cache rate in `VPZ2M7ZP-refs-apse`.

**Comment by glocalglocal** [comment]
Here it is: EEED8UI4-euc

That "?%" is now permanent. I did Export Now several times. caching is 
always on. I edited a source but AE doesn't work with 'when idle' anyway.

**Comment by retorquere** [comment]
Finally something I can reproduce.

**Comment by cjpoor** [comment]
Cached: 100%
Support ID from .2496: YJSR2377-refs-apse

**Comment by glocalglocal** [comment]
same here. Cached: 100%

**Comment by retorquere** [comment]
OK, one down.

**Comment by retorquere** [comment]
Back to tracking down the on-idle problem. This doesn't solve anything, it just adds more logging, so new log ID please.

**Comment by glocalglocal** [comment]
Here it is: B94P797R-euc

**Comment by retorquere** [comment]
@glocalglocal as far as I can tell Zotero never went idle in `B94P797R-euc`, and no auto-exports were performed.

**Comment by glocalglocal** [comment]
5YRZN2AY-euc
How long is idle time? I waited until the screensaver kicked in and the BBT icon is still there waiting.

**Comment by retorquere** [comment]
I have no idea what length of time Zotero waits before it considers the system idle. I just register for the notifier for when it does. `5YRZN2AY-euc` still does not go idle. A google turns up nothing, and the books I own on FF architecture make no mention of it. But I had an error in the time I stacked on top of the idle time. A new build will drop soon that fixes that.

**Comment by glocalglocal** [comment]
I haven't installed .2554 yet but I noticed something with .2551. After I sent you the previous log ID, the BBT icon was still showing. I left my PC for over 90 minutes and when I came back the icon was gone. This may mean BBT idling may have been triggered eventually. When I realised I generated another log ID LYGSSERN-euc during the same Zotero session. It may have captured BBT autoexporting on idle successfully.

**Comment by retorquere** [comment]
Yeah that was the problem I mentioned earlier. I was off by a factor of 1000 for the idle delay because I didn't realize you read the idletime from the idleservice in msecs, but you set it in secs. I thought I had to set it in msecs. You have it set at the default 10, so that works out to 2 hours and approx 45 minutes. 2551 fixes that.

So, export on idle works for you now? The default is a delay of 10 seconds, you can drop that to 1 (but no less) by setting `autoExportIdleWait` in the hidden prefs.

But for @cjpoor, the problem was that it was exporting too frequently, so I'd still like a log from 2551 from you @cjpoor.

**Comment by cjpoor** [comment]
Export occurred after a few minutes with .2554
Support log ID: PEHM46WX-refs-apse

**Comment by cjpoor** [comment]
Automatic export after several minutes with .2551 on Linux Mint
Support log ID with .2551: C5MTFHI6-refs-apse

**Comment by retorquere** [comment]
> > So, export on idle works for you now? The default is a delay of 10 seconds, you can drop that to 1 (but no less) by setting `autoExportIdleWait` in the hidden prefs.
> 
> It works fine. Many thanks. I tried to change the delay. Zotero's Config Editor button does nothing here so I opened `prefs.js` for editing but there is no `autoExportIdleWait` there or anywhere else in that folder and its subfolders. Oh well, one day it will all work as intended.

I realize you probably didn't mean much by this, but I'm away from my computer for half a day so I won't be able to do something about this for a while, so this adds a bit of stress to my day :(

> EDIT I spoke too soon. Now BBT never autoexports. I try to initiate it though the context menu and I got this: 
> 
> ![image](https://user-images.githubusercontent.com/16408198/167025779-df149262-dd9f-475a-a81c-3b5b9c44dddc.png)

This is a different issue, can you open a new issue for this?

**Comment by glocalglocal** [comment]
> I realize you probably didn't mean much by this, but I'm away from my computer for half a day so I won't be able to do something about this for a while, so this adds a bit of stress to my day :(

My point about `autoExportIdleWait` was for information only in case it helps you or someone else. I was just reporting that the setting is unavailable and I was expressing my frustration over Zotero's lack of documentation. I looked around and there is no equivalent preference either. It may be that v6 removed the preference. Nothing to do with you and BBT of course, but I can imagine what it must be like working with undocumented features, and thank you for persevering. 

> This is a different issue, can you open a new issue for this?

Will do.

**Comment by glocalglocal** [comment]
>First auto-export could be uncached and slow, but the cache is retained unless BBT upgrades or a preference that might affect output is changed.

Hmm, it seems BBT deletes its cache even if there is no upgrade and preferences remain unchanged. There may be good reasons, and when it happens it only means that next autoexport is slower. So, no problem there. Thanks.

**Comment by retorquere** [comment]
It really shouldn't delete the cache. If this is about auto exports, you can see the cache rate, it should stay at roughly 100%

**Comment by cjpoor** [comment]
I was able to change the wait time to approx. 120s on Linux Mint (sync turned off).
Support log ID with .2554: LNKRHLUX-refs-apse

Thank you so much for all the hard work fixing this issue, Emiliano. Congratulations!

**Comment by cjpoor** [comment]
The cache has stayed at 100% after two auto exports.

**Comment by retorquere** [comment]
@glocalglocal what is your situation with the cache right now? I'm not sure whether there is still something open.

**Comment by glocalglocal** [comment]
Reported consistently as 100%. No issues here (I think).

**Comment by cjpoor** [comment]
Mine has gone back to ?

**Comment by cjpoor** [comment]
After "Export now" cache is 100%

**Comment by glocalglocal** [comment]
Ah, yes, same here. I see '?' when I run Zotero but when I click on 
'Cached:' that becomes 100%. I was assuming that '?' means BBT doesn't 
bother to check every time unless you ask it to, but when you do, it 
confirms it was 100% already.

**Comment by retorquere** [comment]
It should really update the cache percentage as it exports, and then stay there as new exports are mostly from that cache.

**Comment by glocalglocal** [comment]
Just checked. It does that in addition to what I described earlier (ie 
when I click the 'Cached:' button '?' becomes 100%).

**Comment by cjpoor** [comment]
There was an update of Zotero that would have reset the cache on my machine.

**Comment by cjpoor** [comment]
I got the same behaviour on my Macbook as glocalglocal reported: (ie when I click the 'Cached:' button '?' becomes 100%).

**Comment by cjpoor** [comment]
This also occurs on Linux Mint following a restart (without update) of Zotero (ie when I click the 'Cached:' button '?' becomes 100%). Note: this is instant - does not appear to involve any rebuilding of the cache.

**Comment by cjpoor** [comment]
Sorry. I just realized that I didn't have the latest build on my Macbook (After update of add-ons?). Installed and retesting now.

Noticed on restart of Zotero on Linux that cache is ? and when "Cached:" clicked went to 50%

**Comment by cjpoor** [comment]
Very odd: No auto export occurring (circle stalled) at either "On idle" or "On change" now on Macbook with .2554 installed.
Support log ID: AG3964XV-refs-apse

**Comment by cjpoor** [comment]
Cache was 0%
Did "Export now"
Cached: 100%
Added item to library.
No auto export after 4 minutes "When Idle"
Changed to "On change"
Edited abstract.
No auto export (circle stalled).
Support log ID: FMNYXE95-refs-apse

**Comment by cjpoor** [comment]
Auto export on Macbook has now been done. 
For some reason it took over an hour even though the setting is "On Change."
It is a very large library: .bib file is 14.5MB
Also, I store both the Zotero files and the .bib export on an external hard drive.

**Comment by tophee** [comment]
I am also facing problems with auto export  (on MacOS 12.5.1). It seems stuck on "Scheduled BetterBibTeX" when triggered automatically (i.e. by change):

![Zotero-2022-08-31 at 14 06 12@2x](https://user-images.githubusercontent.com/3662750/187674809-77a604db-7986-49b3-bc4d-b5f13a87069a.png)

But after some time (maybe 15 minutes in my case), it seems to do it. My library is 8.1 MB.

**Comment by retorquere** [comment]
Can't help you without a support log.

**Comment by tophee** [comment]
I just sent the log: C4GFHWR7-euc

**Comment by retorquere** [comment]
You have the auto-export delay set to just under 10 minutes. Given that, I don't see anything surprising in the log. A few changes come in within the 10 minutes , one 5 minutes into the delay, and that sets the new time to export to 10 minutes after that. So 15 minutes sounds about right.
